portfolio/src/main/java/com/google/sps/servlets/NameServlet.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet with a POST handler that updates the name associated with a logged-in user in comments.
+ */
+@WebServlet("/name")
+public class NameServlet extends HttpServlet {
+
+  // Matches names composed of letters, digits, underscores, and spaces between three and twenty characters long.
+  private static final Pattern validName = Pattern.compile("^[\\w ]{3,20}$");
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    String name = request.getParameter("new-name");
+    if (isValidName(name)) {
+      upsertUserInfo(name);
+    }
+  }
+  
+  /**
+   * Checks if the name is composed of only letters, digits, underscores,
+   * and spaces and is between three and twenty characters in length.
+   */
+  private boolean isValidName(String name) {
```

Comment:
If this method doesn't actually require any instance state, you can make this a static method (makes it clear that it does not).